chainstate/src/detail/tokens/mod.rs

```diff
@@ -48,6 +59,65 @@ pub fn check_tokens_burn_data(
     Ok(())
 }
 
+pub fn check_nft_issuance_data(
+    chain_config: &ChainConfig,
+    issuance: &NftIssuanceV1,
+    tx_id: Id<Transaction>,
+    source_block_id: Id<Block>,
+) -> Result<(), TokensError> {
+    // Ticker
+    check_token_text_length(
```

Comment:
So you call a function that gives you a specific error if the length is more than something... and then you override that by map_err and overwriting the error with something else... why not just do the check here in the first place?! What kind of service is that function providing exactly?

Reply:
As I mentioned before, `check_nft_issuance_data` become too big. I tried to improve readability. Probably, better to make a bunch of small functions like `check_nft_something` and remove this layer. What do you think?

Reply:
I'll resolve this. Move the discussion here: https://github.com/mintlayer/mintlayer-core/pull/432#discussion_r987678827